### PR TITLE
Command for listing invites

### PR DESCRIPTION
**Pull request by talavis**
> **Before submitting the PR, please go through the sections below and fill in what you can. If there are any items that are irrelevant for the current PR, remove the row. If a relevant option is missing, please add it as an item and add a PR comment informing that the new option should be included into this template.**

> **All _relevant_ items should be ticked before the PR is merged**

# Description

Add a command `dds user ls --invites` that will use the new endpoint in #1294 in dds_web.

Implements Jira issue DDS-1354.

- [X] Add a summary of the changes and the related issue
- [ ] Add motivation and context regarding why the change is needed
- [ ] List / describe any dependencies or other changes required for this change
- [X] Fixes [link to issue / Jira issue ID]

## Type of change

- [ ] Documentation
- [ ] Workflow
- [ ] Security Alert fix
- [ ] Bug fix (non-breaking)
- [X] New feature (non-breaking)
- [ ] Breaking change (breaking, will cause existing functionality to not work as expected)
- [ ] Tests (only)

# Checklist:

## General

- [x] [Changelog](../CHANGELOG.md): New row added. Not needed when PR includes _only_ tests.
- [X] Code change
  - [X] Self-review of code done
  - [ ] Comments added, particularly in hard-to-understand areas
  - [ ] Documentation is updated

## Repository / Releases

- [ ] Blocking PRs have been merged
- [X] Rebase / update of branch done
- [ ] Product Owner / Scrum Master
  - [ ] The [version](../dds_cli/version.py) is updated (PR to `master` branch)
  - [ ] I am bumping the major version (e.g. 1.x.x to 2.x.x)
    - [ ] I have made the corresponding changes to the API version

## Checks

- [X] Formatting: Black & Prettier checks pass
- [x] CodeQL passes
- [ ] Tests
  - [ ] I have added tests for the new code
  - [ ] The tests pass

### CODE REVIEW

File: dds_cli/account_manager.py [hunk 1]
```diff
@@ -186,42 +186,64 @@ def fix_project_access(self, email, project):
 
         dds_cli.utils.console.print(msg)
 
-    def list_users(self, unit: str = None) -> None:
+    def list_users(self, unit: str = None, invites: bool = None) -> None:
```
Comment (inaod568) [inline dds_cli/account_manager.py:189]: I think this should be split up into two different funcions: 
```
def list_users(self, unit: str = None) -> None:
```
and 
```
def list_invites(self) -> None:
```

File: dds_cli/__main__.py [hunk 2]
```diff
@@ -541,7 +544,7 @@ def list_users(click_ctx, unit):
             no_prompt=click_ctx.get("NO_PROMPT", False),
             token_path=click_ctx.get("TOKEN_PATH"),
         ) as lister:
-            lister.list_users(unit=unit)
+            lister.list_users(unit=unit, invites=invites)
```
Comment (inaod568) [inline dds_cli/__main__.py:547]:
```suggestion
            if invites:
                lister.list_users(unit=unit)
            else: 
                lister.list_invites()
```

**Review by inaod568 — CHANGES_REQUESTED** [review]
Looks good and works as expected but I think there should be a new `list_invites` function instead of adding the if statement to the existing function.

**Review by inaod568 — APPROVED** [review]
Looks good and works as planned - ready to merge after https://github.com/ScilifelabDataCentre/dds_web/pull/1294

**Review by valyo — APPROVED** [review]
Looks fine and works

**Pull request merged**